### ISSUE TITLE
[Suggestion] Install `ttf-symbola` by default

### ISSUE THREAD

**Issue by sarpik** [LukeSmithxyz/LARBS#119]
Hello,

TL;DR:
I'd like to suggest adding `ttf-symbola`s installation to larbs by default.

---

I've had numerous issues with some emojis not being showable (such as st crashing when a file is opened that contains certain emojis) and just recently found this issue https://github.com/LukeSmithxyz/voidrice/issues/232 from voidrice - which actually finally fixed it for me.
 
The steps I took were:

Install `ttf-symbola` from AUR @ https://aur.archlinux.org/packages/ttf-symbola/
(for me it failed when using yay or git, instead I used wget (or just regular download),
then cd /path/to/downloaded/ttf-symbola, tar -xvf ttf-symbola, cd ttf-symbola, makepkg -si, sudo pacman -U generated-package-name)

I suggest this so that other people don't get confused:)

Note - there might be more things that are necessary to completely fix the emoji support such as other fonts (ttf-emojione-color; noto-fonts etc) - they'd probably be useful too!

**Comment by LukeSmithxyz** [comment]
ttf-symbola and ttf-emojione are already installed by the LARBS script by default since 8f21ec51 (December 16, 2018). noto-fonts is also installed as it is pulled as the first valid font dependency.

Since as you say you couldn't install symbola via an AUR helper, this is likely a problem on your end involving your keyrings, permissions or something in your Arch install process. I've just now rebuilt all these packages via the AUR and they work fine.

**Comment by sarpik** [comment]
Great! Yeah, it was probably me, although I just followed your installation videos. Anyways, glad it's fine, have a nice day Luke:)

**Comment by sarpik** [comment]
### Welp - Is there are way to cleanly set up fonts & emojis just like you have in your videos, @LukeSmithxyz? 

---

Okay, shoes. I'm kinda lost with emojis rn. If I have `ttf-symbola` installed, they are available, but they are only black and white. If I uninstall `ttf-symbola`, some become colorful, and some become unavailable (both scenarios are bad).

I've installed arch based on your videos and then used larbs on a clean install to configure everything about a few weeks ago.

I've also tried
* https://github.com/googlei18n/noto-emoji/
* https://www.archlinux.org/packages/extra/any/noto-fonts-emoji/

And failed at https://github.com/googlei18n/noto-emoji/ (the building process doesn't seem to work for me)

It's weird but when I copy an emoji from dmenu (using dmenuunicode) and use `notify-send ((some_emoji))`, the notification shows the emoji correctly just like I want, but `st`, `nvim`, chrome, vscode and other apps don't :cry:
![](https://i.imgur.com/JwmYiYm.jpg)

Dmenu:
![](https://i.imgur.com/1NL9qhy.jpg)

**Comment by LukeSmithxyz** [comment]
Your first problem has to do with what fontconfig is using as the
default font. I can't be sure of your settings. If you want to know what
I have on my system, if I run this:

```
pacman -Qe | grep "font\|ttf"
```

I get these installed fonts:

```
nerd-fonts-complete 2.0.0-5
noto-fonts 20190111-1
noto-fonts-cjk 20181130-2
texlive-fontsextra 2018.50019-1
ttf-emojione 4.5-1
ttf-inconsolata 1:2.0.0.1-3
ttf-junicode 1.002-1
ttf-linux-libertine 5.3.0-4
ttf-ms-fonts 2.0-10
ttf-symbola 12.00-1
```

The texlive, junicode, ms-fonts and nerd-fonts packages shouldn't really
be relevant so you can probably ignore them for your purposes.

I've found that you shouldn't use `ttf-emojione-color` as it actually
tends to default to black and white defaults. Anyway, if you want to
avoid changing fontconfig settings manually, you may want to mimic the
packages I have installed.

Well as for the second question, terminal applications will be using
your monospace black and white font. Everything is working as intended.
They're the same characters. Did you want colorful emojis in the
terminal? If that can be done I've never done it myself.

**Comment by sarpik** [comment]
Strange, I do recall having colorful emojis in st and vim, also in [vscode](https://wiki.archlinux.org/index.php/Visual_Studio_Code), but not anymore :( (I was installing some fonts like `ttf-symbola` and they're no longer colorful:s )

I've now installed the same packages as you just mentioned. Thank you @LukeSmithxyz. If I find a way to have colorful emojis in the terminal I'll for sure let ya know! 

Here's the output of my stuff just in case:

```bash
pacman -Qe | grep "font\|ttf"
```
```console
noto-fonts-emoji 20180810-2
ttf-emojione 4.5-1
ttf-inconsolata 1:2.0.0.1-3
ttf-linux-libertine 5.3.0-4
ttf-symbola 12.00-1
```

---

`$HOME/.Xdefaults`:
```
*.font:	Liberation Mono:pixelsize=22:antialias=true:autohint=true;
```

---

`$HOME/.config/fontconfig/fonts.conf` (copied from voidrice):
```xml
<?xml version='1.0'?>
<!DOCTYPE fontconfig SYSTEM 'fonts.dtd'>
<fontconfig>

 <alias>
    <family>serif</family>
    <prefer><family>Linux Libertine</family></prefer>
  </alias>
  <alias>
    <family>sans-serif</family>
    <prefer><family>Linux Biolinum</family></prefer>
  </alias>
  <alias>
    <family>sans</family>
    <prefer><family>Linux Biolinum</family></prefer>
  </alias>
  <alias>
    <family>monospace</family>
    <prefer><family>Inconsolata</family></prefer>
  </alias>

</fontconfig>
```

**Comment by sarpik** [comment]
# Update

I just came across this issue https://github.com/kovidgoyal/kitty/issues/73

(specifically https://github.com/kovidgoyal/kitty/issues/73#issuecomment-399674797 and https://github.com/kovidgoyal/kitty/issues/73#issuecomment-465984396)

> I suppose this is the maintainer of `AUR`s `ttf-emojione`.

He's been asked how to enable support for emojis in the terminal w/ [kitty](https://github.com/kovidgoyal/kitty) & [ttf-emojione](https://aur.archlinux.org/packages/ttf-emojione).

He says to install the package **AND** to create a symlink for `/etc/fonts/conf.d/75-emojione.conf` to point to `../conf.avail/75-emojione.conf` (which is `/etc/fonts/conf.avail/75-emojione.conf`
(the command is `ln -sf ../conf.avail/75-emojione.conf /etc/fonts/conf.d/75-emojione.conf)

---

After this command I once again have colorful emojis in [vscode](https://wiki.archlinux.org/index.php/Visual_Studio_Code)!!

I'm still trying to mess around with the fonts I have to potentially have colorful support in the `st` terminal (my ego won't let me give up on issue this after wasting 2+ days on it lmao)

**Comment by pat-s** [comment]
Having a very similar problem. Did you figure out a solution meanwhile?

**Comment by sarpik** [comment]
Hey @pat-s - I couldn't manage to get back even uncolored support for my st terminal and realised that probably some configuration files or fonts interfered. It was annoying because I couldn't even use vim properly - emojis made st crash, taking down vim togerher:s

I recently re-installed arch and now emojis work as expected -- although, sadly, they're only black & white.. I'd love a colorful solution, especially for the terminal, but I haven't figured it out yet.

Best of luck trying & if you make it -- please share!:)

**Comment by pat-s** [comment]
I am having exactly the same problem regarding crashes. This also relates to `neomutt` then - if only one message in a mailbox has an emoji, `st` crashes. This makes `neomutt` basically unusable.

I have an idea what might be the reason but need to track it down further. Will let you know.

**Comment by pat-s** [comment]
So the error looks like this:

```sh
X Error of failed request: BadLength (poly request too large or internal Xlib length error)
Major opcode of failed request: 139 (RENDER)
Minor opcode of failed request: 20 (RenderAddGlyphs)
Serial number of failed request: 1442
Current serial number in output stream: 1454 
```

Some interesting resources from the st mailing list:

https://lists.suckless.org/dev/1709/32318.html
https://lists.suckless.org/dev/1709/32325.html

The whole archive can be found here: https://lists.suckless.org/dev/thread.html

TL;DR: Looks like it is not uncommon that `st` crashes with emoji stuff. It is probably unrelated to fonts which is why you (and I) were unable to get it working even by using the same fonts as Luke. I also have `ttf-symbola` installed which fixed the issue for some people.
Still it looks like a different lib is responsible here which may be solved by a system upgrade at some time. 

I couldn't reproduce the error on a Antergos Live system. Currently I am on ArchLabs. It could be that even some distro specific packages are causing the error.

I do not really want to re-install my whole system just because of that. But maybe I have to. Not sure yet..
Guess I'll first try it again in a VM.

**Comment by pat-s** [comment]
## Summary
I did a new installation. But this was not the problem/solution.
I do not know yet who is to blame here. Definitely not Luke or any of the fonts.

The problem is sitting somewhere in `~/.config/`. After deleting all of it, `st` did not crash anymore and emojis worked fine.
Regardless of the font.

Ofc now that my setup is gone I cannot replicate it anymore - However, I still have a second machine with the same "old" setup now that I could debug step by step (by this I mean removing the config files sequentially and checking after each).

**Comment by pat-s** [comment]
I also re-installed machine2. While this worked in the beginning, I am facing the same problem again now.

Machine1 is still fine. I really have no clue...

**Comment by pat-s** [comment]
I am able to reproduce it now. Once I set an emoji font as fallback to have colored emojis in some apps, `st` crashes.

See [here](https://pjs-web.de/post/emoji-support-for-notion-so-on-linux/) how I do it.

Now I need to figure out how to exclude `st` (or more specifically monospace fonts) from the emoji fallback.

**Comment by sarpik** [comment]
Woah, that's a lot you've done - proud of your effort!

I just tried out your article you just linked, and I get the same results - emojis in chrome, vscode & similar apps are colorful now, but `st` crashes, sadly:/

Hoping you'll find a way out for us:D - best of luck once again @pat-s!

**Comment by sarpik** [comment]
Edit - just using this config as local.conf in /etc/fonts gives the same results.
Mentioned in [this reddit comment](https://www.reddit.com/r/archlinux/comments/9q8dlj/how_to_better_enable_color_emojis/efxvrz3?utm_source=share&utm_medium=web2x)

The command is
```bash
sudo curl -o /etc/fonts/local.conf https://pastebin.com/raw/atfUQQ2j
```

**Comment by pat-s** [comment]
Got it. If you remove the following block

```sh
    <match target="pattern">
        <test name="family"><string>monospace</string></test>
        <edit name="family" mode="append"><string>Noto Color Emoji</string></edit>
    </match>
```

from the file linked in my previous post, `st` will not crash.
What this code does is that it sets `noto-fonts-emoji` as a fallback for monospace fonts. This means `st` tries to use this font for eomjis but this won't work as these ones are colored.
`Symbola` is usually used as the default fallback but its explicitly blocked by the following block:

```sh
    <!-- Block Symbola from the list of fallback fonts. -->
    <selectfont>
        <rejectfont>
            <pattern>
                <patelt name="family">
                    <string>Symbola</string>
                </patelt>
            </pattern>
        </rejectfont>
    </selectfont>
```

So when `st` crashes, I assume that in most cases some colorful emoji font is used as a fallback instead of symbola.

There are some other related issues to colored emojis and crashes:

https://github.com/LukeSmithxyz/st/issues/58
https://github.com/LukeSmithxyz/st/issues/37

And more on the `st` mailing list..

**Comment by sarpik** [comment]
Hey, that's interesting. I tried removing the code block and also tried to comment out the `Symbola` blocking block, but my `st` still crashes:S Is there anything else you could've changed? (Probably, I remember trying to fix this myself - you do some many different things that you no longer know what helped and what didn't:D)

**Comment by jpldcarvalho** [comment]
Mine was crashing with same errors due to zsh tmux plugin. You might want to check there too!
